Source/AlchemyLanguageV1/AlchemyLanguageV1.swift

```diff
@@ -0,0 +1,1613 @@
+        let body = try? buildBody(html, html: true)
+        
+        // construct query paramerters
+        var queryParams = [NSURLQueryItem]()
+        
+        queryParams.append(NSURLQueryItem(name: "apikey", value: apiKey))
+        queryParams.append(NSURLQueryItem(name: "outputMode", value: "json"))
+        queryParams.append(NSURLQueryItem(name: "linkedData", value: "1"))
+        if let myUrl = url {
+            queryParams.append(NSURLQueryItem(name: "url", value: myUrl))
+        }
+        if let myGraph = knowledgeGraph {
+            queryParams.append(NSURLQueryItem(name: "knowledgeGraph",
+                value: shouldHaveField(myGraph)))
+        }
+        
+        // construct request
+        let request = RestRequest(
+            method: .POST,
+            url: serviceUrl + "/html/HTMLGetRankedConcepts",
+            contentType: "application/x-www-form-urlencoded",
+            messageBody: body,
+            queryParameters: queryParams
+        )
+        
+        // execute request
+        Alamofire.request(request)
+            .responseObject(dataToError: dataToError) {
+                (response: Response<ConceptResponse, NSError>) in
+                switch response.result {
+                case .Success(let concepts): success(concepts)
+                case .Failure(let error): failure?(error)
+                }
+        }
+        
+    }
+    
+    public func getRankedConceptsText(
+        text: NSURL,
+        knowledgeGraph: Bool? = false,
+        failure: (NSError -> Void)? = nil,
+        success: ConceptResponse -> Void)
+    {
+        // construct body
+        let body = try? buildBody(text, html: false)
+        
+        // construct query paramerters
+        var queryParams = [NSURLQueryItem]()
+        
+        queryParams.append(NSURLQueryItem(name: "apikey", value: apiKey))
+        queryParams.append(NSURLQueryItem(name: "outputMode", value: "json"))
+        queryParams.append(NSURLQueryItem(name: "linkedData", value: "1"))
+        if let myGraph = knowledgeGraph {
+            queryParams.append(NSURLQueryItem(name: "knowledgeGraph",
+                value: shouldHaveField(myGraph)))
+        }
+        
+        // construct request
+        let request = RestRequest(
+            method: .POST,
+            url: serviceUrl + "/text/TextGetRankedConcepts",
+            contentType: "application/x-www-form-urlencoded",
+            messageBody: body,
+            queryParameters: queryParams
+        )
+        
+        // execute request
+        Alamofire.request(request)
+            .responseObject(dataToError: dataToError) {
+                (response: Response<ConceptResponse, NSError>) in
+                switch response.result {
+                case .Success(let concepts): success(concepts)
+                case .Failure(let error): failure?(error)
+                }
+        }
+    }
+    
+    // TODO: Review this. I don't like how large the signature is
```

Comment:
This is not as bad as it appears.  They are optionals with default values so the actual signature varies based on consumer needs.  We could pass in a struct but I want to hold off doing that unless really needed.